packages/model-viewer/src/features/scene-graph/sampler.ts

```diff
@@ -49,29 +49,29 @@ const isWrapMode = (() => {
              wrapModes.indexOf(value as WrapMode) > -1;
 })();
 
-const isValidSamplerValue = <P extends 'minFilter'|'magFilter'|'wrapS'|'wrapT'|'rotation'|'repeat'|'offset'>(
-    property: P, value: unknown): value is DefaultedSampler[P] => {
```

Comment:
I missed this `value is` thing before - I've never seen that syntax. I know it's not yours, but if there's a simpler way, that would be great. Seems like it should just be `P, value: DefaultedSampler<P>): boolean =>`

Reply:
It says DefaultedSampler is not generic. Probably no support for <P>?